### PR TITLE
Decouple help requests

### LINKED ISSUE #19: Decouple 'requests and topics' from RocketChat core - based on new UI
# Context
Currently, all development of the assistify-extension of Rocket.Chat is implemented as a modification.
Though it's separated into packages, there are parts which are implemented as a modification. This is particularly valid for the requests and topics.

# Aim
This issue is about separating all development into a meteor package which is published to meteor package registry: https://atmospherejs.com/

# Current implementation
The package for the requests and topics is `https://github.com/assistify/Rocket.Chat/tree/master/packages/assistify-help-request`
There are no real dependencies to other custom packages, but actually all modifications applied to Rocket.Chat are necessary due to the features implemented in this package.

# Process
In order to do that

- contribute modifications to RC which enable APIs => https://github.com/RocketChat/Rocket.Chat/issues/8246
- Merge the contributions-branch (regularly) to [feature/#19-decouple-help-requests](https://github.com/assistify/Rocket.Chat/tree/feature/%2319-decouple-help-requests) overwriting what's on the target branch
- Get things working again (of course, we'll do unit-tests later ;) )
- Once finished, extract the folder to a separate repo and publish it to meteorpedia
- Delete the folder in this repo

### PR DESCRIPTION
**Pull request by mrsimpson**
Fixes #19 and introduces the new Rocket.Chat UI
**It's not ready yet, this PR is intended for discussions** - at least at the moment until further notice

### CODE REVIEW

File: packages/rocketchat-ui/client/views/app/createChannel.html [hunk 1]
```diff
@@ -1,90 +1,98 @@
 <template name="createChannel">
 	<section class="create-channel">
 		{{#each roomType in roomTypesBeforeStandard}}
-			{{> Template.dynamic template=roomType.creationTemplate }}
+			<div class="room-type-creation">
```
Comment (mrsimpson) [inline packages/rocketchat-ui/client/views/app/createChannel.html]: @ruKurz I wrapped all custom templates into a `div`. This shall allow to visualize them as tabs, I hope

File: packages/rocketchat-ui/client/views/app/createChannel.html [hunk 2]
```diff
@@ -1,90 +1,98 @@
...
-		</header>
-		<form class="create-channel__content">
-			<div class="create-channel__switches">
-				<div class="rc-switch">
-					<label class="rc-switch__label" tabindex="-1">
-						<input type="checkbox" class="rc-switch__input" name="type" value="p" checked>
-						<span class="rc-switch__button">
+		<div class="room-type-creation">
+			<header class="create-channel__header">
+				<h1 class="create-channel__title">{{_ "Create_A_New_Channel"}}</h1>
```
Comment (mrsimpson) [inline packages/rocketchat-ui/client/views/app/createChannel.html]: `.create-channel__header .create-channel__title` as child of `.room-type-creation` could be the title of the tab

**Comment by ruKurz** [comment]
@mrsimpson I've never seen any tabs in Rocket.Chat, what leads me to two questions:
1. Is it smart introducing tabs to an Software as UI element in general (User experience)
2. Does the Rocket.Chat UI framework support tabs in general (as boostrap does)

**Comment by ruKurz** [comment]
I just added a draft for tabs implementation.
It's really not final yet.
@mrsimpson a general feedback would be great, just to make clear, I'm on the right trip.
I added the meteor package: "teamplates:tabs"
https://github.com/meteortemplates/tabs

**Comment by ruKurz** [comment]
IMHO, Slack has solved the creation of Channels, Direct messages, Apps very smooth by adding a "+" Button next to each headline. This is very intuitive, no extra clicks, ...

bildschirmfoto_2017-10-03_um_21_10_04

**Comment by mrsimpson** [comment]
@graywolf336 This branch 

- is based on `develop`
- contains the PR towards Rocket.Chat for the room types (https://github.com/RocketChat/Rocket.Chat/pull/8274)
- utilizes the new API in order to add custom room types

Please review the result and - optionally - implement [what's missing](https://github.com/assistify/Rocket.Chat/pull/75#issuecomment-333619125)

**Comment by mrsimpson** [comment]
@graywolf336 I merged y'day's API changes and adapted the code. It was an enormous help if you could help with the creation-dialogs. I'm quite sure that you'll come up with some ideas for re-factoring it in such a way that Rocketlets providing custom room types can create own UIs on their own more easily.

File: packages/assistify-help-request/startup/customRoomTypes.js [hunk 3]
```diff
@@ -1,135 +1,7 @@
-/* globals openRoom */
+/* globals RocketChat */
```
Comment (graywolf336) [inline packages/assistify-help-request/startup/customRoomTypes.js]: This could be `import {RocketChat} from 'meteor/rocketchat:lib';`

**Comment by mrsimpson** [comment]
Closed this PR as it is completely merged into #97

**Pull request closed without merging**